### PR TITLE
Animations + bear character model

### PR DESCRIPTION
**Pull request by benjxia**
Player now animated when moving around map. Also (slightly) reduced aliasing on outline shader by changing screen shader's sobel operator to roberts-cross operator

### CODE REVIEW

**Comment by Alanine42** [comment]
This assertion in animationPose() always failed when the client started up. 
`assert(animationId >= -1  && animationId < animations.size() && animated);`

I printed the `animationId, animations.size(), animated` and they all look correct. They should not trigger assert false, but they somehow did...

image


IDK what's going on, but after removing the asserts the client ~~runs well~~ looks fabulous!

**Comment by realmatthewpeng** [comment]
I'm getting assert errors as well, in both animationPose and timeToAnimationTick, but my values also look right. Everything looks correct when I comment them out though.

File: client/src/sge/GraphicsGeometry.cpp [hunk 1]
```diff
@@ -329,6 +443,125 @@ namespace sge {
...
+        recursePose(outputModelPose, anim, time, accumulator, bones.root);
+    }
+
+    ModelPose ModelComposite::emptyModelPose() {
+        ModelPose pose(MAX_BONES, glm::mat4(1));
+        return pose;
+    }
+
+    float ModelComposite::timeToAnimationTick(long long milliseconds, int animationId) {
+        assert(animationId >= 0 && animationId < animations.size() && animated);
```
Comment (realmatthewpeng) [inline client/src/sge/GraphicsGeometry.cpp:521]: Why do we assert animationId >= 0 if there is a check on the next line for animationId == -1?

File: client/src/sge/GraphicsGeometry.cpp [hunk 2]
```diff
@@ -329,6 +443,125 @@ namespace sge {
...
+    }
+
+    /**
+     * Return model pose for a given animation at a given timestamp
+     * @param animationId Model's animation identifier
+     * @param time Timestamp to retrieve pose
+     * @return Model's pose at given timestamp
+     */
+    void ModelComposite::animationPose(int animationId, float time, ModelPose& outputModelPose) {
+        assert(animationId >= -1  && animationId < animations.size() && animated);
```
Comment (realmatthewpeng) [inline client/src/sge/GraphicsGeometry.cpp:505]: For some reason, the check animationId < animations.size() is failing for me, even though I have 
animationId = -1
animations.size() = 1

File: client/src/sge/GraphicsGeometry.cpp (hunk 1 above)
Comment (benjxia) [reply client/src/sge/GraphicsGeometry.cpp]: changed to check for animation id >= -1

File: client/src/sge/GraphicsGeometry.cpp (hunk 2 above)
Comment (benjxia) [reply client/src/sge/GraphicsGeometry.cpp]: ye idk wtf is going on here

Comment (Melody-Ruth) [reply client/src/sge/GraphicsGeometry.cpp]: Hmm maybe animations.size() is returning an unsigned int, and animationId is a signed int, and c++ is just casting very strangely?? I will look into this.

Comment (benjxia) [reply client/src/sge/GraphicsGeometry.cpp]: Likely that, if i remember correctly signed ints are converted to unsigned int's in these types of conversions

Comment (Melody-Ruth) [reply client/src/sge/GraphicsGeometry.cpp]: Fixed. Well that's a good reminder for me to think carefully about types, and also run in debug mode occasionally even though it's super slow

**Review by realmatthewpeng — APPROVED** [review]
Now works on my machine! No more assert errors.

**Review by Metarus — APPROVED** [review]
still laggy on my machine because potato but lgtm

**Review by Alanine42 — APPROVED** [review]
lgtm (maybe finetune model position?)

**Comment by Alanine42** [comment]
my bad, just finetuning some player and egg sizes

**Comment by benjxia** [comment]
i dont want to squash this pr, alan has overtaken me in total # of commits cus I've been squashing all my pr's since week 3

**Comment by Alanine42** [comment]
shit i never squash my pr 😂...

**Review by realmatthewpeng — APPROVED** [review]
Still works on my machine

**Pull request merged**